ast/listener.go

```diff
@@ -531,12 +531,19 @@ func (l *listener) ExitWaitBlock(c *parser.WaitBlockContext) {
 // EnvArgKey, EnvArgValue, LabelKey, LabelValue -------------------------------
 
 func (l *listener) EnterEnvArgKey(c *parser.EnvArgKeyContext) {
-	err := checkEnvVarName(c.GetText())
+	text := c.GetText()
+	err := checkEnvVarName(text)
 	if err != nil {
 		l.err = err
 		return
 	}
-	l.stmtWords = append(l.stmtWords, c.GetText())
+	if strings.Contains(text, "'") {
+		text = strings.ReplaceAll(text, "'", "")
```

Comment:
simply removing these seems wrong.

This is valid under docker:

```Dockerfile
FROM alpine
ENV 'really"weird"key' yes
RUN test "$(env | grep weird)" = "really\"weird\"key=yes"
```

even though it's fairly contrived, I think we should maintain this functionality and aim to have earthly on-parity with docker's handling of quoted keys.

Reply:
I tried modifying lex and parse rules in `ast` directory. but was not able to solve this one. shall we just replace if first character is `'` or `"'?

Reply:
I would rather see a generic fix that gets us closer to supporting all dockerfile equivalents.